Sources/tart/Credentials/DockerConfigCredentialsProvider.swift

```diff
@@ -1,16 +1,54 @@
 import Foundation
 
 class DockerConfigCredentialsProvider: CredentialsProvider {
+
   func retrieve(host: String) throws -> (String, String)? {
```

Comment:
Sorry, but this is too much indirection for such a simple function. We now have four new functions containing only 5 lines of code on average, and they're only being called from a single place, thus no code re-use at all.

I think we could get away with just a single new function:

```swift
private func dockerConfig() throws -> Data? { ... }
```

That will either read from `TART_DOCKER_CONFIG` or from the default location, and call it from the `retrieve()` accordingly.